web/src/widgetize.js

```diff
@@ -49,12 +48,8 @@ export function activate(in_opts) {
         });
     };
 
-    if (
-        $row.attr("id").startsWith(`message-row-${message_lists.home.id}-`) &&
-        narrow_state.active()
-    ) {
-        // Don't place widget in a home message row if we are narrowed
-        // to active state
+    if (!$row.attr("id").startsWith(`message-row-${message_lists.current?.id}-`)) {
+        // Don't activate widgets for messages that are not in the current view.
         return;
```

Comment:
I don't love the existing fact that this inspects the DOM so heavily; kinda seems like the caller isn't doing its job to figure out what rows to process, but this migration seems correct in any case.